provisioning/baremetal_pod.go

```diff
@@ -468,10 +476,73 @@ func NewMetal3Deployment(targetNamespace string, images *Images, config *metal3i
 	}
 }
 
-func GetExistingMetal3Deployment(client appsclientv1.DeploymentsGetter, targetNamespace string) (bool, error) {
-	existing, err := client.Deployments(targetNamespace).Get(context.Background(), baremetalDeploymentName, metav1.GetOptions{})
+func annotateExistingMetal3Deployment(existing *appsv1.Deployment) {
+	curAnnotations := existing.GetAnnotations()
+	if _, ok := curAnnotations[cboOwnedAnnotation]; ok {
+		// metal3 deployment already owned by CBO
+		return
+	}
+	cboAnnotations := map[string]string{}
+	cboAnnotations[cboOwnedAnnotation] = ""
+	modified := resourcemerge.BoolPtr(true)
+	resourcemerge.MergeMap(modified, &cboAnnotations, existing.GetAnnotations())
+	existing.SetAnnotations(cboAnnotations)
+	return
+}
+
+func ApplyNewOrUpdatedDeployment(client appsclientv1.DeploymentsGetter, targetNamespace string, metal3Deployment *appsv1.Deployment, generations []osoperatorv1.GenerationStatus, componentName string) error {
+	finalDeployment := metal3Deployment
+
+	// Check if there is an existing metal3 deployment
+	existing, err := client.Deployments(targetNamespace).Get(context.Background(), BaremetalDeploymentName, metav1.GetOptions{})
+
```

Comment:
fyi Zane is adding support for whether or not mao or cbo owns the deployment https://github.com/openshift/cluster-baremetal-operator/pull/72/files#diff-1575ce96065be1a97bee923445ae60115c8ce02b4a2736788012df8162407100R471

Reply:
Yes, I'll rebase this PR after #72 merges.